### DISCUSSION TITLE
DBIP account

### DISCUSSION THREAD

**Discussion by Seb-a320** [wirefalls/geo-nft#31] [Q&A] [answered]
Hello,
Is it possible to use a DBIP account and the non-light version of the CSV?

thx

**Comment by wirefalls** [comment]
> Hello, Is it possible to use a DBIP account and the non-light version of the CSV?
> 
> thx

Hello,

Support for the non-lite version would have to be added to Geolocation for nftables since the format of the CSV file changes between the full and lite versions. The full version adds a two-letter continent code as the third field of each line in the CSV file rather than adding the continent code to the end of the line, so adding support will require some additional work. The login credentials would also need to be added to the geo-nft.conf file. It's been my intention to add this feature, but unfortunately I'm not able to work on it at this time due to other commitments.


Full CSV format:
https://db-ip.com/db/format/ip-to-country/csv.html

Lite CSV format:
https://db-ip.com/db/format/ip-to-country-lite/csv.html

Thanks

**Comment by Seb-a320** [comment, marked as answer]
Hello and thank you for the response. I am not a developer, but in the meantime, this works : 

replacing download_db with

    fichier=`curl -s 'https://db-ip.com/account/xxxxxxxxxxxxxxxxxxxxxxxxxxx/db/ip-to-country/' | jq -r '.csv.name'`
	url=`curl -s 'https://db-ip.com/account/xxxxxxxxxxxxxxxxxxxxxxxxxxx/db/ip-to-country/' | jq -r '.csv.url'`

	wget -O $fichier $url
	gunzip $fichier
	
	while IFS=, read -r f1 f2 f3 f4
	do
	  printf  "$f1,$f2,$f4\n" >> test.csv
	done < "$dbip_filename"

	rm $dbip_filename
	mv test.csv $dbip_filename

**Reply by wirefalls** [reply]
> Hello and thank you for the response. I am not a developer, but in the meantime, this works :
> 
> replacing download_db with
> 
> ```
> fichier=`curl -s 'https://db-ip.com/account/xxxxxxxxxxxxxxxxxxxxxxxxxxx/db/ip-to-country/' | jq -r '.csv.name'`
> url=`curl -s 'https://db-ip.com/account/xxxxxxxxxxxxxxxxxxxxxxxxxxx/db/ip-to-country/' | jq -r '.csv.url'`
> 
> wget -O $fichier $url
> gunzip $fichier
> 
> while IFS=, read -r f1 f2 f3 f4
> do
>   printf  "$f1,$f2,$f4\n" >> test.csv
> done < "$dbip_filename"
> 
> rm $dbip_filename
> mv test.csv $dbip_filename
> ```

That's a nice solution, thank you for sharing the code. When I'm able to work on this I'll add support for the full CSV file. Thanks.
